### PR TITLE
update members and transactions components

### PR DESCRIPTION
**Pull request by yunixon**
https://trello.com/c/fty57MFm/476-transactions-%D0%BF%D0%BE%D0%B4%D1%81%D0%B2%D0%B5%D1%87%D0%B8%D0%B2%D0%B0%D0%B5%D1%82-%D0%BD%D0%BE%D0%B2%D1%8B%D0%B5-%D1%82%D1%80%D0%B0%D0%BD%D0%B7%D0%B0%D0%BA%D1%86%D0%B8%D0%B8

### CODE REVIEW

File: src/components/transactions/list/table-row.tsx [hunk 1]
```diff
@@ -21,16 +22,28 @@ class TransactionsTableRow extends React.Component<Props> {
     }
   }
 
+  getNewTransactionClass = (transaction: Transaction, currentMember: Member): string => {
+    if (transaction.createdAt > currentMember.lastVisitedAt) {
```
Comment (shir) [inline src/components/transactions/list/table-row.tsx]: @yunixon just mark transaction viewed on not on backend. This logic may be used on each client, so there is no reason to make it on client side. And it much easier to implement it in backend. And in this case you will need to load current member.

File: src/components/transactions/list/transactions.tsx [hunk 2]
```diff
@@ -2,23 +2,27 @@ import * as React from 'react';
...
+  orgId:         number;
+  status:        string;
+  transactions:  Transaction[] | null;
+  currentMember: Member;
 }
 
 interface DispatchProps {
-  load: (orgId: number) => void;
+  load:   (orgId: number) => void;
+  update: (orgId: number, memberId: number) => void;
```
Comment (shir) [inline src/components/transactions/list/transactions.tsx]: @yunixon it's bad name because `update` usually mean to update whole object, but this updates only one field.

File: src/components/transactions/list/transactions.tsx [hunk 3]
```diff
@@ -40,10 +44,15 @@ class Transactions extends React.Component<Props> {
     }
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const { orgId, update, currentMember } = this.props;
+    update(orgId, currentMember.id);
```
Comment (shir) [inline src/components/transactions/list/transactions.tsx]: @yunixon not sure if marking `last_visited` in this callback is good because this callback may be called several times (for example when status prop is updated). I think it will be better to call this action in saga after transactions success load.

File: src/api/members.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
-import { prepareURL, getApi } from './_helpers';
+import { prepareURL, getApi, putApi } from './_helpers';
 
-const membersURL = (orgId) => prepareURL(`/api/organizations/${orgId}/members`);
+const membersUrl       = (orgId) => prepareURL(`/api/organizations/${orgId}/members`);
+const organizationUrl  = (orgId) => prepareURL(`/api/organizations/${orgId}`);
```
Comment (shir) [inline src/api/members.js]: @yunixon rename it to `organizationLastVisitUrl` and build full url here.

File: src/components/transactions/list/table-row.tsx [hunk 5]
```diff
@@ -21,6 +21,10 @@ class TransactionsTableRow extends React.Component<Props> {
     }
   }
 
+  getNewTransactionClass = (transaction: Transaction): string => {
```
Comment (shir) [inline src/components/transactions/list/table-row.tsx]: @yunixon it's better to call `rowClass`, so in future it may be used to get all classes for row.

File: src/components/transactions/list/table-row.tsx [hunk 6]
```diff
@@ -30,7 +34,11 @@ class TransactionsTableRow extends React.Component<Props> {
     
     return(
       <>
-        <tr key={ transaction.id } onClick={ () => this.handleClick(transaction.id) }>
+        <tr
+          key={ transaction.id }
```
Comment (shir) [inline src/components/transactions/list/table-row.tsx]: @yunixon no need for `key` attribute here. `key` should be used in upper component.

File: src/components/utils/form-inputs/submit-button.tsx [hunk 7]
```diff
@@ -60,7 +60,7 @@ const HorizontalSubmitButton: React.SFC<Props> = (props) => (
 const VerticalSubmitButton: React.SFC<Props> = (props) => (
   <Row>
     <Col sm={ 12 }>
-      <SubmitButton { ...props } bsStyle="primary pull-right" />
```
Comment (shir) [inline src/components/utils/form-inputs/submit-button.tsx]: @yunixon why did you remove `bsStyle` from here? It's almost always has `primary` style. If you make `<SubmitButton bsStyle="primary" className="pull-right" { ...props } />` (change order for prop) then you will able to override `bsStyle` when you need.

File: src/model-types.ts [hunk 8]
```diff
@@ -18,6 +18,11 @@ export interface Category {
   type:     string;
 }
 
+export interface Member {
```
Comment (shir) [inline src/model-types.ts]: @yunixon Look like it's not used.

File: src/sagas/transactions.js [hunk 9]
```diff
@@ -23,6 +24,7 @@ function* handleLoadTransactions({ payload: { organizationId } }) {
     yield put(loadTransactions.request(organizationId));
     const transactions = yield call(getOrganizationTransactions, organizationId);
     yield put(loadTransactions.success(organizationId, transactions));
+    yield call(putMemberLastViewedAt, organizationId);
```
Comment (shir) [inline src/sagas/transactions.js]: @yunixon instead directly call member API here, call member action which will make update. It will allow to make logic more flexible and call this action from other places in future.

File: src/components/utils/form-inputs/submit-button.tsx (hunk 7 above)
Comment (shir) [reply src/components/utils/form-inputs/submit-button.tsx]: @yunixon notice that `{ ...props}` should be added after other properties so `props` may override default properties.

File: src/components/require-organization/require-organization.jsx [hunk 10]
```diff
@@ -15,33 +15,36 @@ class RequireOrganization extends React.Component {
...
-  addFlashMessage:          PropTypes.func.isRequired,
-  children:                 PropTypes.node,
+  hasOrganization: PropTypes.bool.isRequired,
+  addFlashMessage: PropTypes.func.isRequired,
+  children:        PropTypes.node,
 };
 
 const select = (state) => ({
   hasOrganization: getHasCurrentOrganization(state),
+  orgId:           getCurrentOrganizationId(state),
```
Comment (shir) [inline src/components/require-organization/require-organization.jsx]: @yunixon looks like `orgId` is not used in this component.

File: src/components/transactions/form/form.jsx [hunk 11]
```diff
@@ -24,7 +24,7 @@ const TransactionForm = ({ handleSubmit, type, submitting, error, action, transa
     <Field name="comment" label="Comment" component={ HorizontalFormInput } />
     <Field name="date" label="Date" component={ HorizontalDatePicker } />
     { action === 'Update' && transaction && <DestroyButton transaction={ transaction } /> }
-    <SubmitButton className="pull-right" submitting={ submitting }>{ action } Transaction</SubmitButton>
+    <SubmitButton bsStyle="primary" className="pull-right" submitting={ submitting }>{ action } Transaction</SubmitButton>
```
Comment (shir) [inline src/components/transactions/form/form.jsx]: @yunixon `bsStyle="primary" className="pull-right"` it's default value, not need to use them here.

File: src/components/transactions/form/transfer-form.jsx [hunk 12]
```diff
@@ -25,7 +25,7 @@ const TransferForm = ({ handleSubmit, submitting, error, action, transaction })
     <Field name="comment" label="Comment" component={ HorizontalFormInput } />
     <Field name="date" label="Date" component={ HorizontalDatePicker } />
     { action === 'Update' && transaction && <DestroyButton transaction={ transaction } /> }
-    <SubmitButton className="pull-right" submitting={ submitting }>{ action } Transfer</SubmitButton>
+    <SubmitButton bsStyle="primary" className="pull-right" submitting={ submitting }>{ action } Transfer</SubmitButton>
```
Comment (shir) [inline src/components/transactions/form/transfer-form.jsx]: @yunixon `bsStyle="primary" className="pull-right"` it's default value, not need to use them here.

Comment (yunixon) [reply src/components/transactions/form/transfer-form.jsx]: we have bsStyle="default" as default for SubmitButton, can I change this to 'primary'?

Comment (shir) [reply src/components/transactions/form/transfer-form.jsx]: @yunixon Yes. You can. BTW, why don't we use HorizontalSubmitButton here?
